src/roiextractors/imagingextractor.py

```diff
@@ -68,6 +69,21 @@ def get_dtype(self) -> DtypeType:
     def get_video(self, start_frame: int = None, end_frame: int = None, channel: int = 0) -> np.ndarray:
         return self.get_frames(range(start_frame, end_frame), channel)
 
+    def get_video_shape(self):
+        rows, columns = self.get_image_size()
+        num_channels = self.get_num_channels()
+        unique_frame_value = num_channels + 1
+        if unique_frame_value == rows:
+            unique_frame_value += 1
+        if unique_frame_value == columns:
+            unique_frame_value += 1
+
+        frame_idxs = [_ for _ in range(unique_frame_value)]
+        video_shape = np.array(self.get_frames(frame_idxs=frame_idxs, channel=None).shape)
+        video_shape[video_shape == unique_frame_value] = self.get_num_frames()
+
+        return tuple(video_shape)
```

Comment:
This function is rather strange...

I'd rather have the shape information handled properly by whatever the specific format is so that our existing API functions simply 'handle it' in the same way each time

Reply:
It is written like this to avoid loading as few frames as necessary (for memory concerns). Maybe there is a better way to do it though.

Reply:
Hence why reliance on the format-specific API functions (which should be built lazily as is) would be both simpler and just as safe memory-wise.

See https://github.com/catalystneuro/roiextractors/pull/140#discussion_r901166368